python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/ethernet_interfaces.py

```diff
@@ -176,6 +176,23 @@ def _get_ethernet_interface_cfg(self: AvdStructuredConfigConnectedEndpoints, ada
                 "id": channel_group_id,
                 "mode": port_channel_mode,
             }
+
+            # Update the description to make the port_channel_id usable
+            ethernet_interface["description"] = (
+                self.shared_utils.interface_descriptions.connected_endpoints_ethernet_interface(
+                    InterfaceDescriptionData(
+                        shared_utils=self.shared_utils,
+                        interface=adapter["switch_ports"][node_index],
+                        peer=peer,
+                        peer_interface=peer_interface,
+                        peer_type=connected_endpoint["type"],
+                        description=interface_description,
+                        port_channel_id=channel_group_id,
+                    ),
+                )
+                or None
+            )
```

Comment:
Please incorperate it into the existion description generation instead. You can use `port_channel_id=channel_group_id if port_channel_mode is not None else None` and then move the assignment of port_channel_mode up.